pkg/utils/status.go

```diff
@@ -341,50 +364,67 @@ func calculateState(ctx context.Context, instance *csmv1.ContainerStorageModule,
 	controllerStatus := newStatus.ControllerStatus
 
 	newStatus.State = constants.Failed
-	log.Infof("deployment controllerReplicas [%s]", controllerReplicas)
-	log.Infof("deployment controllerStatus.Available [%s]", controllerStatus.Available)
+	//log.Infof("deployment controllerReplicas [%s]", controllerReplicas)
+	//log.Infof("deployment controllerStatus.Available [%s]", controllerStatus.Available)
 
-	log.Infof("daemonset expected [%d]", expected)
-	log.Infof("daemonset nodeStatus.Available [%s]", nodeStatus.Available)
+	//log.Infof("daemonset expected [%d]", expected)
+	//log.Infof("daemonset nodeStatus.Available [%s]", nodeStatus.Available)
 
-	if instance.GetName() == "application-mobility" {
-		modrunning, err := statusForAppMob(ctx, instance, r, newStatus)
-		if err != nil {
-			log.Infof("statusForAppMob err msg [%s]", err.Error())
+	for _, m := range instance.Spec.Modules {
+		if(m.Name =="application-mobility" && m.Enabled) {
+			appEnabled = true
+			appRunning, err = statusForAppMob(ctx, instance, r, newStatus)
+			if err != nil {
+				log.Infof("status for Application-Mobility err msg [%s]", err.Error())
+			}
 		}
-		if (controllerReplicas == controllerStatus.Available) && (fmt.Sprintf("%d", expected) == nodeStatus.Available) && modrunning {
+
+		if(m.Name == "observability" && m.Enabled) {
+			obsEnabled = true
+			obsRunning, err = observabilityStatusCheck(ctx, instance, r, newStatus)
+			if err != nil {
+				log.Infof("status for Observability err msg [%s]", err.Error())
+			}
+		}
+	}
+
+	if(appEnabled && obsEnabled) {
```

Comment:
@abhi16394 Are app_mobility and Observability interlinked ? Why cannot we make these checks independently and fail the status as and when we see a module failure ?

Reply:
I have changed the code so that it will more reusable in future 
now the scope for dreadnaught is just application-mobility, observability and authorization 
since authorization is deployed as a sidecar container 
we don't need to check that explicitly
we can add for more modules later